src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -90,10 +90,13 @@ public static class EventIds
                 public static readonly EventId ResponseHeader = new EventId(103, "ResponseHeader");
             }
 
-            private static readonly Action<ILogger, HttpMethod, Uri, Exception> _requestStart = LoggerMessage.Define<HttpMethod, Uri>(
+            private static readonly LogDefineOptions _skipEnabledCheckLogDefineOptions = new LogDefineOptions() { SkipEnabledCheck = true };
+
+            private static readonly Action<ILogger, HttpMethod, string, Exception> _requestStart = LoggerMessage.Define<HttpMethod, string>(
                 LogLevel.Information,
                 EventIds.RequestStart,
-                "Sending HTTP request {HttpMethod} {Uri}");
+                "Sending HTTP request {HttpMethod} {AbsoluteUri}",
```

Comment:
```suggestion
                "Sending HTTP request {HttpMethod} {Uri}",
```

While using a relative `Uri` should be uncommon, we decided as part of #55035 to flow it through to the underlying handler and let that throw an appropriate exception. This code shouldn't assume that the Uri is absolute.

I would leave the parameter name as `Uri` and pass in
```c#
_requestStart(logger, request.Method, GetUriString(request.RequestUri), null);

static string? GetUriString(Uri? requestUri)
{
    return requestUri?.IsAbsoluteUri == true
        ? requestUri.AbsoluteUri
        : requestUri?.ToString();
}
```

Reply:
When the request has a relative `Uri` (and no base address set), [HttpClient.PrepareRequestMessage](https://github.com/dotnet/runtime/blob/cf5b231fcbea483df3b081939b422adfb6fd486a/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L767) seems to be called before any of the logging handlers and throws the `'An invalid request URI was provided. Either the request URI must be an absolute URI or BaseAddress must be set.'` exception. Does that make this suggestion still worthwhile?

Reply:
Since `LoggingHttpMessageHandler` is public, a user can wrap their handlers in it without going through `HttpClientFactory`.

While `HttpClient` will perform these checks, `HttpMessageInvoker` will not, so the user could still observe an unexpected exception.

Reply:
Updated 👍